inception-concept-linking/src/main/java/de/tudarmstadt/ukp/inception/conceptlinking/service/ConceptLinkingServiceImpl.java

```diff
@@ -17,754 +17,340 @@
-        }
-        if (!aTypedString.isEmpty()) {
-            Set<CandidateEntity> allCandidates = getCandidatesFullText(aKB, aTypedString);
-            setFrequenciesForCandidates(aKB, allCandidates);
-            // Sort full-text matching candidates by frequency and do cutoff by a threshold
-            List<CandidateEntity> result = sortByFrequency(new ArrayList<>(allCandidates))
-                .stream().limit(properties.getCandidateFrequencyThreshold())
-                .collect(Collectors.toList());
-
-            return result.stream()
-                .map(c -> new KBHandle(c.getIRI(), c.getLabel(), c.getDescription())).distinct()
-                .limit(properties.getCandidateDisplayLimit())
-                .filter(h -> h.getIdentifier().contains(":")).collect(Collectors.toList());
-        }
-        return Collections.emptyList();
-    }
-
-    /**
-     * Remove all cache entries of a specific project
-     * @param aEvent
-     *            The event containing the project
-     */
-    @EventListener
-    public void onKnowledgeBaseConfigurationChangedEvent(
-        KnowledgeBaseConfigurationChangedEvent aEvent)
-    {
-        // FIXME instead of maintaining one global cache, we might maintain a cascaded cache
-        // where the top level is the project and then for each project we have sub-caches.
-        // Then we could invalidate only a specific project's cache. However, right now,
-        // we don't have that and there is no way to properly iterate over the caches and
-        // invalidate only entries belonging to a specific project. Thus, we need to
-        // invalidate all.
-        candidateFullTextCache.invalidateAll();
-        semanticSignatureCache.invalidateAll();
-    }
-
-    private class CandidateCacheKey
-    {
-        private final KnowledgeBase knowledgeBase;
-        private final String query;
-
-        public CandidateCacheKey(KnowledgeBase aKnowledgeBase, String aQuery)
-        {
-            super();
-            knowledgeBase = aKnowledgeBase;
-            query = aQuery;
-        }
-
-
-        public KnowledgeBase getKnowledgeBase()
-        {
-            return knowledgeBase;
-        }
-
-        public String getQuery()
-        {
-            return query;
-        }
-
-        @Override
-        public boolean equals(final Object other)
-        {
-            if (!(other instanceof CandidateCacheKey)) {
-                return false;
-            }
-            CandidateCacheKey castOther = (CandidateCacheKey) other;
-            return new EqualsBuilder().append(knowledgeBase, castOther.knowledgeBase)
-                    .append(query, castOther.query).isEquals();
-        }
-
-        @Override
-        public int hashCode()
-        {
-            return new HashCodeBuilder().append(knowledgeBase).append(query).toHashCode();
-        }
-    }
-
-    private class SemanticSignatureCacheKey
+    public List<KBHandle> searchEntitiesFullText(KnowledgeBase aKB, String aQuery)
```

Comment:
Can you add JavaDoc? I have no idea what this function should do and why a search method is inside the concept linking service, not the search service.

Reply:
Well, the method probably shouldn't be here. I'll leave this for future refactoring.